### ISSUE TITLE
How to remove images created with FromDockerfile?

### ISSUE THREAD

**Issue by dubonzi** [testcontainers/testcontainers-go#303]
I've been creating some of my containers using `FromDockerfile`, is there any way to remove the built images after my tests run? The image list starts to get a bit crowded after a few test runs with all the tags testcontainers create.
![image](https://user-images.githubusercontent.com/795838/112842248-bf4dff80-9077-11eb-8720-456925137672.png)

**Comment by dubonzi** [comment]
> So we can label every image with a specific label that detects the image created by testcontainers in this way users can figure out a way to list and remove the images with a specific label.

This would make removing the images easier. In my case, this is only a problem when running the tests locally, not really a problem, but would be nice to have :). Thanks for the reply!

**Comment by mdelapenya** [comment]
Well it's not actually Java but the test runner you use. We use godog for cucumber scenarios and the test runner provides life cycle hooks for before/after test suite, before/after scenario and before/after step, and there I destroy whatever is needed.

**Comment by xicoalmeida** [comment]
I was having this "problem" and it was kind of boring having to manually delete the containers. 
I don't know if this the best way to solve this but I did it and it's working fine. 

Up til now, my machine didn't explode so... 🤷 

```
t.Cleanup(func() {
			ctx := context.Background()
			cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
			if err != nil {
				panic(err)
			}
			cli.ContainersPrune(ctx, filters.Args{})
		})
```

where `t` is `*testing.T`

**Comment by rzajac** [comment]
I delete images by repo:tag something like this:

```go
func BuildFromDockerfile(ctx context.Context, req tc.ContainerRequest) (string, func(context.Context) error, error) {
	prv, err := tc.NewDockerProvider()
	if err != nil {
		return "", nil, err
	}

	tag, err := prv.BuildImage(ctx, &req)
	if err != nil {
		return "", nil, err
	}

	cleanup := func(ctx context.Context) error {
		cli, err := client.NewClientWithOpts(
			client.FromEnv,
			client.WithAPIVersionNegotiation(),
		)
		if err != nil {
			return err
		}
		opts := types.ImageRemoveOptions{
			Force:         true,
			PruneChildren: true,
		}
		_, err = cli.ImageRemove(ctx, tag, opts)
		return err
	}

	return tag, cleanup, nil
}
```

**Comment by gaborszakacs** [comment]
This custom `BuildFromDockerfile` function (and calling the returned cleanup function in `t.Cleanup` did remove the build image, thanks!

One thing I noticed though is that intermediate containers and images still hang around, I proposed this change to remove them once the build is complete: https://github.com/testcontainers/testcontainers-go/pull/403

What do you think about making image removal getting triggered on `container.Terminate` (if `FromDockerfile` was provided) ?

**Comment by mdelapenya** [comment]
I like this approach @gaborszakacs, and it's something I was discussing with @kiview the other day. If this is something you can provide in a PR, I would be more than grateful to review it. Thanks in advance!

**Comment by gaborszakacs** [comment]
Thanks @mdelapenya, for the input, I gave it a shot: https://github.com/testcontainers/testcontainers-go/pull/405

**Comment by gaborszakacs** [comment]
@mdelapenya gentle ping, these two PRs addresses this issue, can you take a look at them pls?
- https://github.com/testcontainers/testcontainers-go/pull/403
https://github.com/testcontainers/testcontainers-go/pull/405

**Comment by mdelapenya** [comment]
I think we can close this one, as above PRs have been merged. Thank you all for your feedback
